clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/ops/servergroup/DeployKubernetesAtomicOperation.groovy

```diff
@@ -93,31 +93,40 @@ class DeployKubernetesAtomicOperation implements AtomicOperation<DeploymentResul
     }
 
     task.updateStatus BASE_PHASE, "Replica set name chosen to be ${replicaSetName}."
+    def hasDeployment = KubernetesApiConverter.hasDeployment(description)
+    def replicaSet
 
     if (description.source?.useSourceCapacity) {
-      task.updateStatus BASE_PHASE, "Reading ancestor server group ${description.source.serverGroupName}..."
-      def ancestorServerGroup = credentials.apiAdaptor.getReplicationController(description.source.namespace, description.source.serverGroupName)
+      task.updateStatus BASE_PHASE, "Searching for ancestor server group ${description.source.serverGroupName}..."
+      def ancestorServerGroup = credentials.apiAdaptor.getReplicationController(namespace, description.source.serverGroupName)
       if (!ancestorServerGroup) {
-        ancestorServerGroup = credentials.apiAdaptor.getReplicaSet(description.source.namespace, description.source.serverGroupName)
+        ancestorServerGroup = credentials.apiAdaptor.getReplicaSet(namespace, description.source.serverGroupName)
```

Comment:
Is it possible that the source might exist in some other namespace? It seems like that's something that should be supported.

Reply:
Thanks for pointing this out.

Reply:
@lwander what about the case where it might not be set you because the Server Group is dynamic?  That was causing issues for me when testing a workflow we have...

Reply:
I would default to the current namespace, but rely on source when it's available

Reply:
Sounds good.